sdk/storage/inc/blobs/blob_service_client.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "blob_client_options.hpp"
+#include "common/storage_credential.hpp"
+#include "common/storage_url_builder.hpp"
+#include "internal/protocol/blob_rest_client.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace Blobs {
+
+  class BlobServiceClient {
+  public:
+    // connection string
+    static BlobServiceClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    // shared key auth
+    explicit BlobServiceClient(
+        const std::string& serviceUri,
+        std::shared_ptr<SharedKeyCredential> credential,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    // token auth
+    explicit BlobServiceClient(
+        const std::string& serviceUri,
+        std::shared_ptr<TokenCredential> credential,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    // anonymous/SAS/customized pipeline auth
+    explicit BlobServiceClient(
+        const std::string& serviceUri,
+        const BlobServiceClientOptions& options = BlobServiceClientOptions());
+
+    ListContainersSegment ListBlobContainersSegment(
+        const ListBlobContainersOptions& options = ListBlobContainersOptions()) const;
+
+    UserDelegationKey GetUserDelegationKey(
```

Comment:
Why would UserDelegationKey contain below members? They don't seem to be related to UserDelegationKey.
```
    std::string RequestId;
    std::string Date;
    std::string Version;
    std::string ClientRequestId;
```

Reply:
Because we don't have `Response<T>` yet, these members should be put in `Response<T>`.

Reply:
I don't agree with this. I think the Response<T> will only have two things:
1. `GetResponse()`: This API gets the raw response from the request.
2. `GetValue()`: This API gets the `<T>` value parsed by Storage SDK's protocol layer.
We cannot force `Response<T>` to have these four members since the type is shared across service while the four response headers are only used in storage.

Reply:
once `Response<T>` is ready, i'll remove these four fields from every API response. But now, if I removed it, the user had no way to check the `client-request-id`.

Reply:
I don't think this should be removed. Customers should be able to easily get those values without looking into the raw response.
This can be discussed in the future.
Also, please consider the E2E scenario for this. Customers should be able to use a UserDelegationKey to generate a UserDelegationSas, which makes this return value an input value for some of our API, which will be more bizarre.

Reply:
@katmsft We can add these APIs in the future

Reply:
I created an issue https://github.com/Azure/azure-sdk-for-cpp/issues/176 to track this